pkg/api/findings.go

```diff
@@ -93,8 +93,16 @@ func validateFindingTransition(finding model.FindingExpanded, payload model.Patc
 	// FALSE_POSITIVE -> OPEN
 	// OPEN           -> FALSE_POSITIVE
 	// FALSE_POSITIVE -> FALSE_POSITIVE
-	if (finding.Status != "OPEN" && finding.Status != "FALSE_POSITIVE") ||
-		(payload.Status != "OPEN" && payload.Status != "FALSE_POSITIVE") {
+	// FIXED          -> FALSE_POSITIVE
+	if finding.Status == payload.Status {
+		return nil
+	}
+	validTransitions := map[string]any{
+		"OPEN->FALSE_POSITIVE":  nil,
+		"FALSE_POSITIVE->OPEN":  nil,
+		"FIXED->FALSE_POSITIVE": nil,
+	}
+	if _, valid := validTransitions[fmt.Sprintf("%s->%s", finding.Status, payload.Status)]; !valid {
 		return echo.NewHTTPError(http.StatusConflict, fmt.Sprintf("Status transition not allowed: from '%s' to '%s'", finding.Status, payload.Status))
 	}
```

Comment:
I would change how we check that the transition from status.
I would do something like this
```suggestion
validTransitions := map[string]string{
		"OPEN":           "FALSE_POSITIVE",
		"FALSE_POSITIVE": "OPEN",
		"FIXED":          "FALSE_POSITIVE",
	}
	if allowed := validTransitions[finding.Status]; allowed == payload.Status {
		return echo.NewHTTPError(http.StatusConflict, fmt.Sprintf("Status transition not allowed: from '%s' to '%s'", finding.Status, payload.Status))
	}
```

Reply:
That would work for this case, but it won't support transitions to more than one state like this:

OPEN->FALSE_POSITIVE
OPEN->FIXED

Reply:
Yes. What you did it's indeed the default solution, I didn't suggested it because initially we don't need to support those transitions, but that's okey.